### DISCUSSION TITLE
Does watchtower actually downloads the image or just detects if image is updated?

### DISCUSSION THREAD

**Discussion by urza** [containrrr/watchtower#1471] [Q&A] [answered]
Hi, sorry if this is stupid question, but I would like to know if the watchtower is actually downloading the image from registry or if it is able to use hash/version/something to detect the image in registry is updated and download it only in such case?

Thanks

**Comment by piksel** [comment, marked as answer]
Yes, the current containers image hash is compared with one that is fetched from the repository using a HEAD request. And then, only if those hashes doesn't match, it pulls the new image.
